src/Nox.Types/Types/Image/Image.cs

```diff
@@ -0,0 +1,134 @@
+    /// <param name="options">The <see cref="ImageTypeOptions"/> containing constraints for the value object.</param>
+    /// <returns>An instance of the <see cref="Image"/> class.</returns>
+    public static Image From(string url, string prettyName, int sizeInBytes, ImageTypeOptions options)
+    {
+        var newObject = new Image
+        {
+            Value = (url, prettyName, sizeInBytes),
+            _imageTypeOptions = options
+        };
+        
+        var validationResult = newObject.Validate();
+
+        if (!validationResult.IsValid)
+        {
+            throw new TypeValidationException(validationResult.Errors);
+        }
+
+        return newObject;
+    }
+
+    /// <summary>
+    /// Retrieves the equality components for the image value object.
+    /// </summary>
+    /// <returns>An enumerable collection of key-value pairs representing the equality components.</returns>
+    protected override IEnumerable<KeyValuePair<string, object>> GetEqualityComponents()
+    {
+        yield return new KeyValuePair<string, object>(nameof(Url), Url);
+        yield return new KeyValuePair<string, object>(nameof(PrettyName), PrettyName);
+        yield return new KeyValuePair<string, object>(nameof(SizeInBytes), SizeInBytes);
+    }
+
+    /// <summary>
+    /// Returns a string representation of the image.
+    /// </summary>
+    /// <returns>A string representation of the image.</returns>
+    public override string ToString()
+    {
+        return $"{PrettyName}: {Url}";
+    }
+
+    /// <summary>
+    /// Validates the <see cref="Image"/> object.
+    /// </summary>
+    /// <returns>A validation result indicating whether the <see cref="Image"/> object is valid or not.</returns>
+    internal override ValidationResult Validate()
+    {
+        var result = base.Validate();
+
+        if (string.IsNullOrWhiteSpace(Value.Url))
+        {
+            result.Errors.Add(new ValidationFailure(nameof(Value), $"Could not create a Nox Image type with an empty Url."));
+        }
+        else if (!Uri.TryCreate(Value.Url, UriKind.Absolute, out _))
+        {
+            result.Errors.Add(new ValidationFailure(nameof(Value), $"Could not create a Nox Image type with an invalid Url '{Value.Url}'."));
+        }
+        else if (!_imageTypeOptions.GetSupportedImageFormatTypeExtensions().Contains(Path.GetExtension(Value.Url).ToLower()))
+        {
+            result.Errors.Add(new ValidationFailure(nameof(Value), $"Could not create a Nox Image type with an image having an unsupported extension '{Value.Url}'."));
+        }
+
+        if (string.IsNullOrWhiteSpace(Value.PrettyName))
+        {
+            result.Errors.Add(new ValidationFailure(nameof(Value), $"Could not create a Nox Image type with an empty PrettyName."));
+        }
+
+        if (Value.SizeInBytes <= 0)
+        {
+            result.Errors.Add(new ValidationFailure(nameof(Value), $"Could not create a Nox Image type with a Size of {Value.SizeInBytes}."));
```

Comment:
Should the message be $"Could not create a Nox Image type with a **negative** Size of {Value.SizeInBytes}."?